src/views/Home.jsx

```diff
@@ -1,22 +1,72 @@
+import { useEffect } from 'react';
+import { useState } from 'react';
 import { generateToken } from '@the-collab-lab/shopping-list-utils';
-import { useNavigate } from 'react-router-dom';
+import { streamListItems } from '../api';
 import './Home.css';
 
 export function Home({ handleListTokenState }) {
-	const navigate = useNavigate();
+	const [userEnteredToken, setUserEnteredToken] = useState('');
+	const [message, setMessage] = useState('');
+
+	useEffect(() => {
+		const timer = setTimeout(() => {
+			setMessage(null);
+		}, 3000);
+
+		return () => clearTimeout(timer);
+	}, [message]);
 
 	const handleCreateList = () => {
 		const newToken = generateToken();
 		handleListTokenState(newToken);
-		navigate('/list');
+	};
+
+	const handleChange = (e) => setUserEnteredToken(e.target.value);
+	const handleFormSubmit = (e) => {
+		e.preventDefault();
+		streamListItems(userEnteredToken, (snapshot) => {
```

Comment:
It'd be a good idea here to check if there's no token and display an appropriate message.

Reply:
Resolved! Nice catch!